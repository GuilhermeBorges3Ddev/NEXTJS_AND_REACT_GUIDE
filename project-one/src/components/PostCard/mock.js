export const PostCardPropsMock = {
  title: 'Some title',
  body: 'Description',
  id: 1,
  cover: 'img/img.png',
};
