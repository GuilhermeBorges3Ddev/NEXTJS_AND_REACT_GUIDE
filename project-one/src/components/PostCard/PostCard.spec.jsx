import React from "react";
/* eslint-disable testing-library/no-node-access */
import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { PostCardPropsMock } from "./mock";

const props = PostCardPropsMock;

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    /* 
        If you wanna debug:
            const { debug } = render(<PostCard {...PostCardPropsMock} />);
            debug();
    */
    render(<PostCard {...props} />);
    const headingText = `${props.title} - ${props.id}`;
    const imageElements = screen.getAllByRole("img", { name: props.title });
    const headingElements = screen.getAllByRole("heading", {
      name: headingText,
    });
    const paragraphText = screen.getByText(props.body);
    expect.assertions(3);
    expect(imageElements[0]).toHaveAttribute("src", props.cover);
    expect(headingElements[0]).toBeInTheDocument();
    expect(paragraphText).toBeInTheDocument();
  });
  it("should match a snapshot of the PostCard tested above", () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
