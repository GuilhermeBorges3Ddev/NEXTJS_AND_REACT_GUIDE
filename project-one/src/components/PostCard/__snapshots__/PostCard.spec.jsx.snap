// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match a snapshot of the PostCard tested above 1`] = `
<div
  class="post"
>
  <img
    alt="Some title"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Some title
       - 
      1
    </h2>
    <p>
      Description
    </p>
  </div>
</div>
`;
